pkg/osvscanner/osvscanner.go

```diff
@@ -864,7 +890,7 @@ func makeRequest(
 		case p.PURL != "":
 			query.Queries = append(query.Queries, osv.MakePURLRequest(p.PURL))
 		default:
-			return nil, fmt.Errorf("package %v does not have a commit, PURL or ecosystem/name/version identifier", p)
+			log.Panic("packages received in makeRequest should have already been filtered")
```

Comment:
nit: this panic message is a little tied to the implementation details of osvscanner as a whole -- i.e. this assumes knowledge about the caller's behaviour. 

Should we just return an error here insteaD?

Reply:
Good point, changed back to returning  the error